firmware/src/src.ino

```diff
@@ -118,53 +132,70 @@ void serialEvent() {
     // so the main loop can do something about it:
     if (inChar == '\n') {
       stringComplete = true;
+      receivedFirstMessage = true; // first contact! (handshaking)
+      //send_warning("debug Arduino serialEvent complete", message);
       return;
     }
   }
 }
 // #endregion
 
+
+//-----------------------------------------------------------------------------
 void actuatorLoop(){
   // If serial message, actuate based on it.
-  if(stringComplete){
-    String splitMessages[COMMAND_LENGTH];
-    for(int i = 0; i < COMMAND_LENGTH; i++){
-      splitMessages[0] = "";
```

Comment:
fun bug here!   `splitMessages[0]` is changed to `[i]` farther down in the diff.